evm/db/account.py

```diff
@@ -301,24 +333,44 @@ def _get_account(self, address):
 
     def _set_account(self, address, account):
         rlp_account = rlp.encode(account, sedes=Account)
-        self._trie[address] = rlp_account
-        cache_key = (id(self._unwrapped_db), self.state_root, address)
-        account_cache[cache_key] = rlp_account
+        self._journaltrie[address] = rlp_account
 
     #
     # Record and discard API
     #
-    def record(self) -> UUID:
-        return self.db.record()
+    def record(self) -> Tuple[UUID, UUID]:
+        return (self._journaldb.record(), self._journaltrie.record())
 
-    def discard(self, changeset_id: UUID) -> None:
-        return self.db.discard(changeset_id)
+    def discard(self, changeset: Tuple[UUID, UUID]) -> None:
+        db_changeset, trie_changeset = changeset
+        self._journaldb.discard(db_changeset)
+        self._journaltrie.discard(trie_changeset)
 
-    def commit(self, changeset_id: UUID) -> None:
-        return self.db.commit(changeset_id)
+    def commit(self, changeset: Tuple[UUID, UUID]) -> None:
+        db_changeset, trie_changeset = changeset
+        self._journaldb.commit(db_changeset)
+        self._journaltrie.commit(trie_changeset)
 
     def persist(self) -> None:
-        return self.db.persist()
-
-    def clear(self) -> None:
-        return self.db.reset()
+        self._journaldb.persist()
+        self.logger.debug("Persisting account values...")
+        accounts_displayed = set()  # type: Set[bytes]
+        queued_changes = self._journaltrie.journal.journal_data.items()
+        # mypy bug for ordered dict reversibility: https://github.com/python/typeshed/issues/2078
+        for checkpoint, accounts in reversed(queued_changes):  # type: ignore
+            for address in accounts:
+                if address in accounts_displayed:
+                    continue
+                else:
+                    accounts_displayed.add(address)
+                    account = self._get_account(address)
+                    self.logger.debug(
+                        "Saving Account %s: balance %d, nonce %d, storage root %s, code hash %s",
+                        encode_hex(address),
+                        account.balance,
+                        account.nonce,
+                        encode_hex(account.storage_root),
+                        encode_hex(account.code_hash),
+                    )
```

Comment:
If it was pure logging, then I'd be good with it, but the loop and whatnot doesn't sit well with me.  Maybe if you moved this outside the function body into a stand-alone and left it inline but commented out so that it's easy to enable as a one-liner but it doesn't clutter up the actual function body.